### ISSUE TITLE
WalletConnect does not stay active if phone is locked, goes to sleep, or the app put in the background. Connected site thinks WalletConnect is still active.

### ISSUE THREAD

**Issue by sk33z3r** [Loopring/loopring-wallet-feedback#167]
**Phone Details**:
* Android 11
* Loopring Wallet v2.1.3.179

**Repro**:
1. Open wallet on Android and enter PIN to login
2. Open Loopring web wallet
3. Use WalletConnect to connect
_Observe_: Mobile app presented with a signing message with an "Approve" button
4. Let the phone go to sleep, manually press the lock button on the phone, or go to the home screen
5. Try to unlock the wallet in the web UI

**Expected**:
When I click UNLOCK in the web UI, it asks me to approve a message in the wallet app. Upon opening the wallet app I am presented with the approval message I need to sign in order to unlock the wallet in the web UI.

**Actual**:
After opening the wallet app, I am presented with the normal landing screen, but it no longer indicates that I am connected to any sites. The web UI still thinks I am still connected.

---

**Other Notes**:
My specific repro here isn't too big of an issue, but this does become an issue when you are working with something like ENS. If I register a new ENS, then want to edit the record and add a bunch of data, my phone usually goes to sleep by the time I am done entering all the new information into the records. Then I try to run the CONFIRM operation on ENS, but when I go to approve the message I never get it in the mobile app. It's just disconnected, but ENS still says that my wallet is connected regardless.

In order to get things to work, I have to then disconnect on the ENS site, refresh the page, reconnect using WalletConnect, and then while I am entering information into the records on ENS, I have to constantly go back to my phone and tap the screen just to keep it awake until I get all my info pasted and ready to hit confirm.

**Comment by e2l33** [comment]
I have this same problem. Disconnects every time my phone closes

**Comment by sk33z3r** [comment]
If some fix can't be found for this particular issue, maybe some option in settings to "Keep the phone awake while open" would be an option.

**Comment by e2l33** [comment]
is that an option on iphone? I've never seen it.  But maybe could be coded into the LRC app.  I'm not knowledgeable about coding.

**Comment by sk33z3r** [comment]
@e2l33 I know it's possible on Android, but not sure for iOS. Figured I would place the comment as food for thought for the devs.

**Comment by sk33z3r** [comment]
The new behavior that I am seeing seems to be improved slightly, but the underlying issue is still here. Fairly consistently, I have this new behavior:

* First, understand there's a different between phones going to "sleep", getting "locked", and the screen going off.
* I believe that the behavior is resolved for the instances where my phone is only locked and not yet asleep, or the screen is only off and phone not yet locked.
* If I allow my phone to sit long enough for the phone to actually sleep/go into power save mode, then the app tends to close itself and I get the original issue 100% of the time so far still.

**Comment by sk33z3r** [comment]
This issue seems to be improving more and more as the day goes on. I just had my first semi-successful unlock after my phone went to sleep for about an hour. After I tried to unlock this is what went down:

1. Wake up phone and unlock it
2. Open Loopring Wallet (it closed itself at some point while the phone was asleep)
_OBSERVE_: I see the WalletConnect icon in the bottom-right
3. Click "Unlock" in the web UI
_OBSERVE_: I receive the message to approve the unlock
4. Tap Approve on the phone

**Expected**:
Message is signed and my wallet unlocked in the web UI

**Actual**:
Unlock modal shows a loading gif and never unlocks the wallet. Android phone still claims WalletConnect is active.

I had to disconnect and refresh the page to get connected again.
